app/controllers/users_controller.rb

```diff
@@ -7,8 +7,8 @@ def new
   def create
     @user = User.new(user_params)
     if @user.save
-      flash[:success] = "You are now registered and logged in"
-      session[:user] = @user
+      flash[:success] = "You are now registered and logged in as #{@user.name}"
+      session[:user_id] = @user.id
```

Comment:
Had this slightly wrong, easy fix. Added name to flash message.

Reply:
Perfect! I thought this was needed. Glad to see it worked.